codesigngroup.go

```diff
@@ -0,0 +1,199 @@
+	var macCodeSignGroup *export.MacCodeSignGroup
+	exportProfileMapping := map[string]string{}
+
+	// We do not need provisioning profile for the export if the app in the generated XcArchive doesn't
+	// contain embedded provisioning profile.
+	if archive.Application.ProvisioningProfile != nil {
+		installedCertificates, err := certificateutil.InstalledCodesigningCertificateInfos()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get installed certificates: %s", err)
+		}
+		certificates := certificateutil.FilterValidCertificateInfos(installedCertificates)
+		validCertificates := append(certificates.ValidCertificates, certificates.DuplicatedCertificates...)
+
+		log.Debugf("\n")
+		log.Debugf("Installed valid certificates:")
+		for _, certInfo := range validCertificates {
+			log.Debugf(certInfo.String())
+		}
+
+		log.Debugf("\n")
+		log.Debugf("Installed invalid certificates:")
+		for _, certInfo := range certificates.InvalidCertificates {
+			log.Debugf(certInfo.String())
+		}
+		log.Debugf("\n")
+
+		installedProfiles, err := profileutil.InstalledProvisioningProfileInfos(profileutil.ProfileTypeMacOs)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get installed provisioning profiles: %s", err)
+		}
+
+		log.Debugf("\n")
+		log.Debugf("Installed profiles:")
+		for _, profInfo := range installedProfiles {
+			log.Debugf(profInfo.String())
+		}
+
+		var validInstallerCertificates []certificateutil.CertificateInfoModel
+		if exportMethod == exportoptions.MethodAppStore {
+			installedInstallerCertificates, err := certificateutil.InstalledInstallerCertificateInfos()
+			if err != nil {
+				log.Errorf("Failed to read installed Installer certificates, error: %s", err)
+			}
+			installerCertificates := certificateutil.FilterValidCertificateInfos(installedInstallerCertificates)
+			validInstallerCertificates = append(installerCertificates.ValidCertificates, installerCertificates.DuplicatedCertificates...)
+
+			log.Debugf("\n")
+			log.Debugf("Installed valid installer certificates:")
+			for _, certInfo := range validInstallerCertificates {
+				log.Debugf(certInfo.String())
+			}
+
+			log.Debugf("\n")
+			log.Debugf("Installed invalid installer certificates:")
+			for _, certInfo := range installerCertificates.InvalidCertificates {
+				log.Debugf(certInfo.String())
+			}
+		}
+
+		macCodeSignGroup, err = matchingMacCodeSignGroups(archive, validCertificates, validInstallerCertificates, installedProfiles, exportMethod, teamID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find code sign groups for the project: %s", err)
+		}
+
+		if macCodeSignGroup != nil {
+			for bundleID, profileInfo := range macCodeSignGroup.BundleIDProfileMap() {
+				exportProfileMapping[bundleID] = profileInfo.Name
+			}
+		}
+
+	} else {
+		log.Printf("Archive was generated without provisioning profile.")
+		log.Printf("Export the application using automatic signing...")
```

Comment:
Is it called automatic signing? As far as I see, It is not automatic, just does not need the bundleID - profile mapping.